### PR TITLE
Add toBeOdd matcher

### PR DESCRIPTION
**Pull request by soueuls**
### What
toBeOdd

### Why
Implement #32 

### Housekeeping

- [x] Unit tests
- [x] Documentation is up to date
- [x] No additional lint warnings
- [x] Add yourself to contributors list (`yarn contributor`)

### CODE REVIEW

File: src/matchers/toBeOdd/index.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import each from 'jest-each';
+
+import matcher from './';
+
+expect.extend(matcher);
+
+describe('.toBeOdd', () => {
```
Comment (mattphillips) [inline src/matchers/toBeOdd/index.test.js]: The failing case is missing for `.toBeOdd`

```
  each([
    [false],
    [true],
    [''],
    [2],
    [[]],
    [{}],
    [() => {}],
    [undefined],
    [null],
    [NaN]
  ]).it('fails when not given an odd number: %s', given => {
    expect(() => expect(given).toBeOdd()).toThrowErrorMatchingSnapshot();
  });
```

File: src/matchers/toBeOdd/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { matcherHint, printExpected, printReceived } from 'jest-matcher-utils';
+
+import predicate from './predicate';
+
+const passMessage = received => () =>
+  matcherHint('.not.toBeOdd', 'received', '') + '\n\n' + 'Expected value to not be odd received:\n' + `  ${received}`;
```
Comment (mattphillips) [inline src/matchers/toBeOdd/index.js]: The `received` value should be wrapped in the printReceived function. `${printReceived(received)};`

File: src/matchers/toBeOdd/index.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+const failMessage = received => () =>
+  matcherHint('.toBeOdd', 'received', '') +
+  '\n\n' +
+  'Expected value to be odd:\n' +
```
Comment (mattphillips) [inline src/matchers/toBeOdd/index.js]: This should be:
```
'Expected value to be odd received:\n' + `  ${printReceived(received)}`;
```

**Comment by mattphillips** [comment]
Looks good just a few minor comments :smile: let me know if you need any help

**Comment by soueuls** [comment]
Thanks for the quick feedback, I will happily make those changes tomorrow.

I don't need any help with it, that's trivial to fix and was just stupid mistakes I made. :disappointed:

**Comment by mattphillips** [comment]
Good job @soueuls thanks a lot for the contribution :smile: don't forget to star the project

**Pull request merged**